### PR TITLE
[release-1.26] Backport misc fixes from upstream.

### PR DESCRIPTION
**Pull request by flouthoc**
Backports fixes from `upstream` (`1.27.0-dev`) to `release-1.26`:
https://github.com/containers/buildah/pull/4013 squash, layers: never use build cache when computing last step of last stage
https://github.com/containers/buildah/pull/4086 add: fail on bad HTTP response instead of writing to container for URL sources
https://github.com/containers/buildah/pull/3998 run: allow re-using secret artifact from host in new RUN steps

### CODE REVIEW

**Comment by flouthoc** [comment]
@TomSweeneyRedHat PTAL backported fixes which comes without any complications.

**Comment by rhatdan** [comment]
LGTM

**Review by vrothberg — APPROVED** [review]
/lgtm

**Comment by flouthoc** [comment]
@TomSweeneyRedHat @rhatdan Is there anything else that we need to backport ? or should we cut `1.26.3` now.

**Comment by rhatdan** [comment]
I am waiting for your approval.  I am fine to go with what we have as long as the containers/common is updated.

**Comment by TomSweeneyRedHat** [comment]
I'm late to the review party, but LGTM and thanks @flouthoc !

**Pull request merged**
